insights/tests/client/test_client.py

```diff
@@ -24,63 +48,212 @@ def test_version():
+        InsightsConfig(register=True, analyze_container=True)
+
+
+def test_unregister_container():
+    with pytest.raises(ValueError):
+        InsightsConfig(unregister=True, analyze_container=True)
+
+
+def test_force_reregister_container():
+    with pytest.raises(ValueError):
+        InsightsConfig(reregister=True, analyze_container=True)
+
+
+def test_reg_check_registered():
+    # register the machine first
+    config = InsightsConfig()
+    client = InsightsClient(config)
+    client.connection = FakeConnection(registered=True)
+    client.session = True
+
+    # test function and integration in .register()
+    assert client.get_registration_information()['status'] is True
+    assert client.register() is True
+    for r in constants.registered_files:
+        assert os.path.isfile(r) is True
+    for u in constants.unregistered_files:
+        assert os.path.isfile(u) is False
+
+
+def test_reg_check_unregistered():
+    # unregister the machine first
+    config = InsightsConfig()
+    client = InsightsClient(config)
+    client.connection = FakeConnection(registered='unregistered')
+    client.session = True
+
+    # test function and integration in .register()
+    assert client.get_registration_information()['status'] is False
+    assert client.register() is False
+    for r in constants.registered_files:
+        assert os.path.isfile(r) is False
+    for u in constants.unregistered_files:
+        assert os.path.isfile(u) is True
+
+
+def test_reg_check_registered_unreachable():
+    # register the machine first
+    config = InsightsConfig(register=True)
+    client = InsightsClient(config)
+    client.connection = FakeConnection(registered=None)
+    client.session = True
+    assert client.register() is True
+
+    # reset config and try to check registration
+    config.register = False
+    client.connection = FakeConnection(registered=False)
+    assert client.get_registration_information()['unreachable'] is True
+    assert client.register() is None
+    for r in constants.registered_files:
+        assert os.path.isfile(r) is True
+    for u in constants.unregistered_files:
+        assert os.path.isfile(u) is False
+
+
+def test_reg_check_unregistered_unreachable():
+    # unregister the machine first
+    config = InsightsConfig(unregister=True)
+    client = InsightsClient(config)
+    client.connection = FakeConnection(registered=True)
+    client.session = True
+    assert client.unregister() is True
+
+    # reset config and try to check registration
+    config.unregister = False
+    client.connection = FakeConnection(registered=False)
+    assert client.get_registration_information()['unreachable'] is True
+    assert client.register() is None
+    for r in constants.registered_files:
+        assert os.path.isfile(r) is False
+    for u in constants.unregistered_files:
+        assert os.path.isfile(u) is True
+
+
+# @patch('insights.client.client.constants.sleep_time', 0)
```

Comment:
If you cherry-pick my changes to these tests, it won’t be necessary to comment them out. They resolve merge conflicts.

https://github.com/Glutexo/insights-core/commit/f10ffed5b4e399dc22e99d0e95f0f3adcae015a7